### PR TITLE
Add Canvas Connect

### PR DESCRIPTION
**Pull request by fauxbytes**

### CODE REVIEW

File: packages/config/src/layer2s/canvasconnect.ts [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+    },
+    activityDataSource: 'Closed API',
+  },
+  config: {
+    escrows: [],
+    transactionApi: {
+      type: 'starkex',
+      product: 'canvasconnect',
+      sinceTimestamp: new UnixTime(1675465200),
+      resyncLastDays: 7,
```
[the remark this answers is not in the document]
Comment (fauxbytes) [reply packages/config/src/layer2s/canvasconnect.ts]: How about [now](https://github.com/CanvasDeFi/l2beat/commit/6838d479e8dcfeb4cfcdc56fdd48ffb1782e04c2)?

File: packages/config/src/layer2s/canvasconnect.ts [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+        'https://canvasdefi.medium.com/',
+        'https://www.linkedin.com/company/canvasblockchaingroup',
+        'https://www.canvas.co/content',
+      ],
+    },
+    activityDataSource: 'Closed API',
+  },
+  config: {
+    escrows: [],
+  },
```
Comment (michalsidzej) [inline packages/config/src/layer2s/canvasconnect.ts:46]: I'm slightly confused by this. The project will always have 0 TVL unless we manually add the L2 native tokens. Also, please explain, how would the DEFI pooling work without any L1 assets.

File: packages/config/src/layer2s/canvasconnect.ts [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+export const canvasconnect: Layer2 = {
+  type: 'layer2',
+  id: ProjectId('canvasconnect'),
+  display: {
+    name: 'Canvas Connect',
+    slug: 'canvasconnect',
+    description:
+      'Canvas Connect is a privacy focused Layer 2 ZK network built for Finance and the Digital Economy. Connect offers an API first model that supports easy integration with no solidity skills required. Capital Markets Applications are built on the network that enable the Transfer, Trade & Investment of Tokenised Real World Assets, Stablecoins, CBDCs & Digital Assets with assurances of Privacy, Low Cost and High Speed, all secured by Ethereum.',
+    purpose:
+      'Transfer, Trade & Investment of Tokenised Real World Assets, Stablecoins, CBDCs & Digital Assets with assurances of Privacy, Low Cost and High Speed, all secured by Ethereum.',
```
[the remark this answers is not in the document]
Comment (michalsidzej) [reply packages/config/src/layer2s/canvasconnect.ts:27]: The `purpose` should be short as it goes here 
Screenshot 2023-03-22 at 12 08 30

File: packages/config/src/layer2s/canvasconnect.ts [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+  STATE_CORRECTNESS,
+} from './common'
+import { Layer2 } from './types'
+
+export const canvasconnect: Layer2 = {
+  type: 'layer2',
+  id: ProjectId('canvasconnect'),
+  display: {
+    name: 'Canvas Connect',
+    slug: 'canvasconnect',
```
Comment (michalsidzej) [inline packages/config/src/layer2s/canvasconnect.ts:23]:
```suggestion
    slug: 'canvasconnect',
    warning: 'Canvas Connect is focused on institutional clients and now there is no possibility for an individual user to use this L2.',
```

File: packages/config/src/layer2s/canvasconnect.ts [hunk 5]
```diff
@@ -0,0 +1,153 @@
...
+        sinceTimestamp: new UnixTime(1675209600),
+        tokens: ['ETH','USDC'],
+      },
+
+    ],
+  },
+  riskView: makeBridgeCompatible({
+    stateValidation: RISK_VIEW.STATE_ZKP_ST,
+    dataAvailability: RISK_VIEW.DATA_EXTERNAL_DAC,
+    upgradeability: RISK_VIEW.UPGRADE_DELAY('14 days'),
```
Comment (michalsidzej) [inline packages/config/src/layer2s/canvasconnect.ts:57]:
```suggestion
    upgradeability: RISK_VIEW.UPGRADE_YES,
```

File: packages/config/src/layer2s/canvasconnect.ts [hunk 6]
```diff
@@ -0,0 +1,153 @@
...
+      },
+      {
+        name: 'Committee',
+        address: EthereumAddress('0x8B3A6662809195453645e37C2005d655f57ca818'),
+        description:
+          'Data Availability Committee (DAC) contract verifying data availability claim from DAC Members (via multisig check).',
+      },
+      SHARP_VERIFIER_CONTRACT,
+    ],
+    risks: [CONTRACTS.UPGRADE_WITH_DELAY_RISK('14 days')],
```
Comment (michalsidzej) [inline packages/config/src/layer2s/canvasconnect.ts:95]:
```suggestion
    risks: [CONTRACTS.UPGRADE_NO_DELAY_RISK],
```

File: packages/config/src/layer2s/canvasconnect.ts [hunk 7]
```diff
@@ -0,0 +1,153 @@
...
+      repositories: ['https://github.com/starkware-libs/starkex-contracts'],
+      socialMedia: [
+        'https://twitter.com/canvas_defi',
+        'https://www.youtube.com/@canvas_defi',
+        'https://canvasdefi.medium.com/',
+        'https://www.linkedin.com/company/canvasblockchaingroup',
+        'https://www.canvas.co/content',
+      ],
+    },
+    activityDataSource: 'Closed API',
```
Comment (michalsidzej) [inline packages/config/src/layer2s/canvasconnect.ts:42]:
```suggestion
```

File: packages/config/src/layer2s/canvasconnect.ts [hunk 8]
```diff
@@ -0,0 +1,153 @@
...
+} from './common'
+import { Layer2 } from './types'
+
+export const canvasconnect: Layer2 = {
+  type: 'layer2',
+  id: ProjectId('canvasconnect'),
+  display: {
+    name: 'Canvas Connect',
+    slug: 'canvasconnect',
+    warning: 'Canvas Connect is initially open to whitelisted institutional clients. Access for individuals will follow at a later date.',
```
Comment (michalsidzej) [inline packages/config/src/layer2s/canvasconnect.ts:24]:
```suggestion
    warning: 'Canvas Connect is initially open only to whitelisted institutional clients.',
```

**Pull request merged**